cli/dcoscli/auth/main.py

```diff
@@ -67,8 +73,84 @@ def _info(info):
     return 0
 
 
-def _login():
+def _verify_cert(cert, yes):
+    """
+    User verification of cert
+
+    :param cert:
+    :returns: verification of cert
+    :rtype: bool
+    """
+
+    pem_cert = load_certificate(FILETYPE_PEM, cert)
+    fingerprint = pem_cert.digest("sha1")
+    msg = ("Correct sha1 fingerprint of cluster certificate?: "
+           "{}".format(fingerprint.decode('utf-8')))
+    return confirm(msg, yes=yes)
+
+
+def _add_cluster_cert(dcos_url, yes):
+    """
+    If the user allows, download the cluster CA and set `core.dcos_url`
+    to the path of the downloaded CA
+
+    :param dcos_url: cluster url
+    :type dcos_url: str
+    :param yes: whether or not to prompt user for confirmation
+    :type yes: bool
+    :returns: whether or not successfully added cert
+    :rtype: bool
+    """
+
+    msg = "An SSL error occured. Download the cluster CA cert?"
+    if not confirm(msg, yes=yes):
+        return False
+
+    with util.temptext() as file_tmp:
+        _, cert_tmp = file_tmp
+        ca_cert_url = urllib.parse.urljoin(dcos_url, "ca/api/v2/info")
+
+        with open(cert_tmp, 'wb') as f:
+            r = http.post(ca_cert_url, verify=False, json={})
+            cert = r.json()["result"].get("certificate").encode('utf-8')
+            if _verify_cert(cert, yes):
+                f.write(cert)
+            else:
+                return False
+
+            # store the cert in DCOS_DIR
+            dcos_dir = os.path.expanduser(
+                os.path.join("~", constants.DCOS_DIR))
+            util.ensure_dir_exists(dcos_dir)
+            cert_path = os.path.join(dcos_dir, "dcos-ca.pem")
+            shutil.move(cert_tmp, cert_path)
+            # add cert to config
+            config.set_val("core.ssl_verify", cert_path)
```

Comment:
This solution only allows you to have one certificate per user. Some user have more than one configuration file.